saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/assertion/SAMLAttributeValueParser.java

```diff
@@ -105,7 +105,23 @@ public Object parse(XMLEventReader xmlEventReader) throws ParsingException {
             return StaxParserUtil.getElementText(xmlEventReader);
         }
 
-        throw logger.parserUnknownXSI(typeValue);
+        // KEYCLOAK-18417: Simply ignore unknown types
+        logger.warn("Skipping attribute value of unsupported type " + typeValue);
+        skipElementAndSubElements(xmlEventReader);
```

Comment:
Please leverage existing code, there is already `StaxParserUtil.bypassElementBlock(xmlEventReader)` for this which does not suffer from potential infinite loop (see below).

Reply:
Changed this according to your suggestion. Just a remark: The method `StaxParserUtil.bypassElementBlock(xmlEventReader)`uses exactly the same mechanism that I proposed, the only difference being that it implements  incrementing and decrementing of a counter instead of recursion (the result is the same, though, honestly I do not see how either implementation could lead to infinite loops).

Reply:
> Changed this according to your suggestion. Just a remark: The method `StaxParserUtil.bypassElementBlock(xmlEventReader)`uses exactly the same mechanism that I proposed, the only difference being that it implements incrementing and decrementing of a counter instead of recursion (the result is the same, though, honestly I do not see how either implementation could lead to infinite loops).

Thank you. When the SAML document contains some other entities, e.g. comments or characters, the `xmlEventReader.peek()` would keep peeking the same event indefinitely. To get the event consumed, at some point, `xmlEventReader.nextEvent()` must be called. See https://docs.oracle.com/javase/8/docs/api/javax/xml/stream/events/XMLEvent.html for all possible events, some of which might appear in the body of that method.